tests/unauthorized/filesTest.js

```diff
@@ -36,77 +36,19 @@ fdescribe('Unauthorized: Currently testing files management,', function () {
 
   beforeAll(function (done) {
     const promises = []
+    const requiredMethodsForUnauthorizedTest = ['GET', 'PUT', 'MKCOL', 'DELETE', 'COPY', 'MOVE', 'PROPFIND']
     promises.push(setGeneralInteractions(provider))
-    promises.push(provider.addInteraction({
-      uponReceiving: 'a request to get file contents with invalid authentication',
-      withRequest: {
-        method: 'GET',
-        path: webdavUrl,
-        headers: requestHeaderWithInvalidAuth
-      },
-      willRespondWith: unauthorizedResponse
-    }))
-    promises.push(provider.addInteraction({
-      uponReceiving: 'a request to put file contents with invalid authentication',
-      withRequest: {
-        method: 'PUT',
-        path: webdavUrl,
-        headers: requestHeaderWithInvalidAuth
-      },
-      willRespondWith: unauthorizedResponse
-    }))
-    promises.push(provider.addInteraction({
-      uponReceiving: 'a request to delete file contents with invalid authentication',
-      withRequest: {
-        method: 'DELETE',
-        path: webdavUrl,
-        headers: requestHeaderWithInvalidAuth
-      },
-      willRespondWith: unauthorizedResponse
-    }))
-    promises.push(provider.addInteraction({
-      uponReceiving: 'a request to create a directory with invalid authentication',
-      withRequest: {
-        method: 'MKCOL',
-        path: webdavUrl,
-        headers: {
-          authorization: invalidAuthHeader,
-          Origin: origin
-        }
-      },
-      willRespondWith: {
-        status: 401,
-        headers: webdavFilesResponseHeader,
-        body: incorrectAuthorizationXmlResponseBody
-      }
-    }))
-    promises.push(provider.addInteraction({
-      uponReceiving: 'a request to list files with invalid authentication',
-      withRequest: {
-        method: 'PROPFIND',
-        path: webdavUrl,
-        headers: requestHeaderWithInvalidAuth
-      },
-      willRespondWith: unauthorizedResponse
-    }))
-    promises.push(provider.addInteraction({
-      uponReceiving: 'a request to move file with invalid authentication',
-      withRequest: {
-        method: 'MOVE',
-        path: webdavUrl,
-        headers: requestHeaderWithInvalidAuth
-      },
-      willRespondWith: unauthorizedResponse
-    }))
-    promises.push(provider.addInteraction({
-      uponReceiving: 'a request to copy a file with invalid authentication',
-      withRequest: {
-        method: 'COPY',
-        path: webdavUrl,
-        headers: requestHeaderWithInvalidAuth
-      },
-      willRespondWith: unauthorizedResponse
-    }))
+    requiredMethodsForUnauthorizedTest.forEach((method) => {
+      promises.push(provider.addInteraction({
+        uponReceiving: `a ${method} request on file contents with invalid authentication`,
+        withRequest: {
+          method: method,
+          path: webdavUrl,
+          headers: requestHeaderWithInvalidAuth
+        },
+        willRespondWith: unauthorizedResponse
+      }))
+    })
```

Comment:
```
provider.addInteraction({
      uponReceiving: 'a request to copy a file with invalid authentication',
      withRequest: {
        method: Pact.Matchers.term({
  matcher: 'GET|PUT|MKCOL|DELETE|COPY|MOVE|PROPFIND',
  generate: 'COPY'
}),
        path: webdavUrl,
        headers: requestHeaderWithInvalidAuth
      },
      willRespondWith: unauthorizedResponse
    })
```

Reply:
no loop needed

Reply:
I've also tried to use regex to match `method`. But Pact is not happy.
![Screenshot from 2020-10-06 19-00-38](https://user-images.githubusercontent.com/39373750/95206398-614a0900-0806-11eb-91e8-6b2087cee872.png)

Reply:
sorry I haven't read your comment above